### PR TITLE
New3ds

### PR DESCRIPTION
**Pull request by uziwuzzy**
1. implement new 3ds flow
2. enable 3ds config from snaptoken
3. fix credit card installment when required
4. remove gopay app detection, allowing to transact directly on sandbox only.

### CODE REVIEW

**Comment by femmerling** [comment]
lgtm

File: corekit/src/main/java/com/midtrans/sdk/corekit/core/MidtransApiService.java [hunk 1]
```diff
@@ -19,12 +19,11 @@
      *
      * @return callback of transaction response
      */
-    @GET("token")
+    @GET("token?x_source=android")
```
Comment (pahlevikun) [inline corekit/src/main/java/com/midtrans/sdk/corekit/core/MidtransApiService.java]: Will be better if this endpoint not containing hardcoded query.

File: demo/src/main/AndroidManifest.xml [hunk 2]
```diff
@@ -12,43 +13,45 @@
...
         <activity android:name=".DemoProductPageActivity" />
         <activity
             android:name=".DemoOrderReviewActivity"
             android:windowSoftInputMode="stateHidden|adjustPan" />
         <activity
             android:name=".UserAddressActivity"
-            android:windowSoftInputMode="stateHidden|adjustPan">
-
-        </activity>
+            android:windowSoftInputMode="stateHidden|adjustPan"></activity>
```
Comment (pahlevikun) [inline demo/src/main/AndroidManifest.xml]: Better to close directly if there's no item inside `activity` like  `xxxx|adjustPan"/>`

File: demo/src/main/res/layout/activity_entry.xml [hunk 3]
```diff
@@ -0,0 +1,9 @@
+<?xml version="1.0" encoding="utf-8"?>
+<android.widget.LinearLayout xmlns:android="http://schemas.android.com/apk/res/android"
+    xmlns:app="http://schemas.android.com/apk/res-auto"
+    xmlns:tools="http://schemas.android.com/tools"
+    android:layout_width="match_parent"
+    android:layout_height="match_parent"
+    tools:context="com.midtrans.demo.EntryActivity">
```
Comment (pahlevikun) [inline demo/src/main/res/layout/activity_entry.xml]: Unused layout? EntryActivity not use this

File: uikit/src/main/java/com/midtrans/sdk/uikit/views/gopay/payment/GoPayPaymentActivity.java [hunk 4]
```diff
@@ -33,6 +33,8 @@
 
     private static final String TAG = GoPayPaymentActivity.class.getSimpleName();
     private final String GOJEK_PACKAGE_NAME = "com.gojek.app";
+    private final String GOJEK_DEEPLINK_PREFIX_OLD = "gojek://";
```
Comment (pahlevikun) [inline uikit/src/main/java/com/midtrans/sdk/uikit/views/gopay/payment/GoPayPaymentActivity.java]: Better to use static variable

File: corekit/src/main/java/com/midtrans/sdk/corekit/core/MidtransApiService.java (hunk 1 above)
Comment (farhanpahlevi) [inline corekit/src/main/java/com/midtrans/sdk/corekit/core/MidtransApiService.java]: Will be better if this endpoint not containing hardcoded query.

File: demo/src/main/AndroidManifest.xml (hunk 2 above)
Comment (farhanpahlevi) [inline demo/src/main/AndroidManifest.xml]: Better to close directly if there's no item inside activity like `xxxx|adjustPan"/>`

File: demo/src/main/res/layout/activity_entry.xml [hunk 5]
```diff
@@ -0,0 +1,9 @@
+<?xml version="1.0" encoding="utf-8"?>
```
Comment (farhanpahlevi) [inline demo/src/main/res/layout/activity_entry.xml]: Unused layout? EntryActivity not use this

File: uikit/src/main/java/com/midtrans/sdk/uikit/views/gopay/payment/GoPayPaymentActivity.java (hunk 4 above)
Comment (farhanpahlevi) [inline uikit/src/main/java/com/midtrans/sdk/uikit/views/gopay/payment/GoPayPaymentActivity.java]: Better to use static variable

**Pull request merged**